core/src/test/java/nl/nn/adapterframework/jdbc/JdbcTransactionalStorageTest.java

```diff
@@ -0,0 +1,201 @@
+				+ "CONSTRAINT PK_IBISSTORE PRIMARY KEY("+storage.getKeyField()+"));"); 
+	}
+
+	@After
+	public void teardown() throws Exception {
+		if (tableCreated) {
+			JdbcUtil.executeStatement(connection, "DROP TABLE "+tableName); // drop the table if it was created, to avoid interference with Liquibase
+		}
+	}
+
+	@Test
+	public void testQueryTextAndBrowseMessage() throws Exception {
+		testQueryTextAndBrowseMessageHelper(true);
+	}
+
+	@Test
+	public void testQueryTextAndBrowseMessageNotCompressed() throws Exception {
+		testQueryTextAndBrowseMessageHelper(false);
+	}
+
+	public void testQueryTextAndBrowseMessageHelper(boolean blobsCompressed) throws Exception {
+		
+		storage.setBlobsCompressed(blobsCompressed);
+		storage.configure();
+
+		storage.createQueryTexts(dbmsSupport);
+		// check created query
+		String expected = "SELECT "+keyField+","+messageField+" FROM "+tableName+" WHERE "+keyField+"=?";
+		String query = storage.selectDataQuery; 
+		assertEquals(expected, query);
+
+		Message message = Message.asMessage("message");
+		StringBuilder sb = new StringBuilder();
+		for(int i=0; i<5;i++) {
+			sb.append(message);
+		}
+
+		// insert a record 
+		PreparedStatement stmt = prepareStatement();
+
+		ByteArrayOutputStream baos = new ByteArrayOutputStream(); 
+		OutputStream out = blobsCompressed ? new DeflaterOutputStream(baos) : baos;
+		try (ObjectOutputStream oos = new ObjectOutputStream(out)) {
+			oos.writeObject(message);
+		}
+		stmt.setBytes(1, baos.toByteArray());
+		stmt.execute();
+		// check inserted data being correctly retrieved
+		Message data =  storage.browseMessage("1");
+		assertEquals(message.asString(), data.asString());
+
+	}
+
+	@Test
+	public void testRetriveObject() throws Exception {
+		testRetriveObjectHelper(true);
+	}
+
+	@Test
+	public void testRetrieveObjectNotCompressed() throws Exception {
+		testRetriveObjectHelper(false);
+	}
+
+	public void testRetriveObjectHelper(boolean blobsCompressed) throws Exception {
+		storage.setBlobsCompressed(blobsCompressed);
+		storage.configure();
+
+		Message message = createMessage();
+
+		// insert a record 
+		PreparedStatement stmt = prepareStatement();
+
+		ByteArrayOutputStream baos = new ByteArrayOutputStream(); 
+		OutputStream out = blobsCompressed ? new DeflaterOutputStream(baos) : baos;
+		try (ObjectOutputStream oos = new ObjectOutputStream(out)) {
+			oos.writeObject(message);
+		}
+		stmt.setBytes(1, baos.toByteArray());
+		stmt.execute();
+
+		ResultSet rs = connection.prepareStatement("SELECT * FROM "+tableName).executeQuery();
+
+		if(rs.next()) {
+			Message result = storage.retrieveObject(rs, 9);
+			assertEquals(message.asString(),result.asString());
+		}
+	}
+
+	private PreparedStatement prepareStatement() throws SQLException {
+		return connection.prepareStatement("INSERT INTO "+tableName+" VALUES"
+				+ "(1,'E','test','localhost','messageId','correlationId','2021-07-13 11:04:19.860','comments',?,'2021-07-13 11:04:19.860','label')");
```

Comment:
Does this run for all databases, including MSSQL? I would expect dbmsSupport.getDatetimeLiteral() to be used here.